manifests/backend.pp

```diff
@@ -113,6 +113,8 @@
       if ('httpchk' in $options['option']) {
         warning('Overriding the value of $sort_options_alphabetic to "false" due to "httpchk" option defined')
         $_sort_options_alphabetic = false
+      } else {
+        $_sort_options_alphabetic = $picked_sort_options_alphabetic
       }
     } else {
       $_sort_options_alphabetic = $picked_sort_options_alphabetic
```

Comment:
Crap, I can't suggest the code change because it is larger that the scope GitHub allows to suggest changes for, but I wonder if we can make this more readable with a single type matching test:

```puppet
  } else {
    if $options =~ Struct[{ option => Struct[{ httpchk => Any }] }] {
      warning('Overriding the value of $sort_options_alphabetic to "false" due to "httpchk" option defined')
      $_sort_options_alphabetic = false
    } else {
      $_sort_options_alphabetic = $picked_sort_options_alphabetic
    }
```

Reply:
thanks, applied it!

Reply:
D'oh, `in` can check if a key exist in a Hash and if a value exist in an Array, and we had both usages here, which I failed to spot.  Your original fix was good, and my suggestion cannot be applied with this code, sorry about that!

Reply:
ah right. I added my old code back.